### PR TITLE
e2e: adding incentivized interchain accounts bank send test

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Adding incentivized interchain accounts test using `banktypes.MsgSend` for happy path scenario

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: e2e/interchain_accounts_test.go
Comment (damiannolan) [inline e2e/interchain_accounts_test.go:21]: How is this import grouping for e2e? Is this okay, or should we separate ibc-go imports?

Comment (charleenfei) [reply e2e/interchain_accounts_test.go:21]: i dont have a strong opinion, but separation seems clearer to me

**Review by charleenfei — APPROVED** [review]
adding my approving review, although since i also worked on this branch we should have at least one more reviewer, cc @seantking

File: e2e/interchain_accounts_test.go
Comment (chatton) [inline e2e/interchain_accounts_test.go:83]: Think these are duplicates of the functions belonging to `FeeMiddlewareTestSuite`

If they are needed in two test suites, maybe they should go into `testsuite.E2ETestSuite`?

Either that or we could extract a type that has these functions and compose both of the other test suites of that type.

File: e2e/testsuite/grpc_query.go
Comment (chatton) [inline e2e/testsuite/grpc_query.go:31]: this function was going to be used in one of @colin-axner 's coming tests I believe. I don't think we should remove this.

**Review by chatton — COMMENTED** [review]
Looks good just a few small comments.

I think we can hold off for my PR to be merged and then change the base branch to `main`.

File: e2e/testsuite/grpc_query.go
Comment (damiannolan) [reply e2e/testsuite/grpc_query.go:30]: Wow... I actually have no idea how this got removed. I never touched it 🤣 
I'll put it back!

File: e2e/interchain_accounts_test.go
Comment (damiannolan) [reply e2e/interchain_accounts_test.go:83]: Yeah, I agree. Do you think we should do this in a follow up PR?

**Review by chatton — APPROVED** [review]
LGTM nice job!

**Review by seantking — APPROVED** [review]

File: e2e/interchain_accounts_test.go
Comment (colin-axner) [inline e2e/interchain_accounts_test.go:46]: I think we should put all incentivized tests under one test suite. Since ics29 was added in v4 but ics27 was added in v3. We will want to do base ics27 tests using v3 for cross compatibility purposes, but incentivization tests should only occur as far back as v4

File: e2e/interchain_accounts_test.go
Comment (colin-axner) [inline e2e/interchain_accounts_test.go:353]: for simplicity, I wonder if we should just make this a suite function so it matches `s.StartRelayer(relayer)`

File: e2e/interchain_accounts_test.go
Comment (colin-axner) [inline e2e/interchain_accounts_test.go:259]: Why was the packet relay assertion removed? Without knowing that the packet was relayed, we can't be certain that the balance didn't change because a failed transfer or because the packet wasn't relayed since we are checking the receiving balance and not the sender

**Pull request merged**
